### PR TITLE
ETFE-2208: Create Import Customs Office Page

### PR DESCRIPTION
**Pull request by r-melvin**

### CODE REVIEW

File: app/views/sections/importInformation/ImportCustomsOfficeCodeView.scala.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        @govukInput(
+            InputViewModel(
+                field = form("value"),
+                label = LabelViewModel(HtmlContent(
+                    if(userType == GreatBritainRegisteredConsignor || (userType == NorthernIrelandRegisteredConsignor && dispatchPlace == GreatBritain)) {
+                        messages("importCustomsOfficeCode.greatBritainRegisteredConsignor.heading")
+                    } else {
+                        messages("importCustomsOfficeCode.northernIrishRegisteredConsignor.heading")
+                    }
+                )).asPageHeading(LabelSize.Large)
```
Comment (LadonJackson) [inline app/views/sections/importInformation/ImportCustomsOfficeCodeView.scala.html:81]: I think the heading being here is causing it to be displayed in the wrong place

File: conf/app.routes [hunk 2]
```diff
@@ -34,4 +34,4 @@ GET         /trader/:ern/draft/:draftId/confirmation         controllers.Confirm
 ->          /                                            exportInformation.Routes
 ->          /                                            destination.Routes
 ->          /                                            guarantor.Routes
-->          /                                            exportInformation.Routes
+->          /                                            importInformation.Routes
```
Comment (LadonJackson) [inline conf/app.routes:37]: This probably wasn't intentional

Comment (r-melvin) [reply conf/app.routes:37]: It should be importInformation, exportInformation is already on line 34

File: app/controllers/sections/importInformation/ImportCustomsOfficeCodeController.scala [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                                                  override val userAnswersService: UserAnswersService,
+                                                  formProvider: ImportCustomsOfficeCodeFormProvider,
+                                                  val controllerComponents: MessagesControllerComponents,
+                                                  view: ImportCustomsOfficeCodeView
+                                                 ) extends BaseNavigationController with AuthActionHelper {
+
+  def onPageLoad(ern: String, draftId: String, mode: Mode): Action[AnyContent] =
+    authorisedDataRequestAsync(ern, draftId) {
+      implicit request =>
+        withAnswer(DispatchPlacePage) {
```
Comment (james-work-account) [inline app/controllers/sections/importInformation/ImportCustomsOfficeCodeController.scala:51]: Neither XIRC nor GBRC get asked the Dispatch Place question. Can you double check with the UCD folk if the text in the ticket is actually correct? I don't see any mention of the Dispatch Place in the copydeck, it only says "if XIRC then text is ABC, if GBRC then text is DEF".

File: app/controllers/sections/importInformation/ImportCustomsOfficeCodeController.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+              userType,
+              dispatchPlace
+            )))
+        }
+    }
+
+  def onSubmit(ern: String, draftId: String, mode: Mode): Action[AnyContent] =
+    authorisedDataRequestAsync(ern, draftId) {
+      implicit request =>
+        withAnswer(DispatchPlacePage) {
```
Comment (james-work-account) [inline app/controllers/sections/importInformation/ImportCustomsOfficeCodeController.scala:65]: As above

File: app/controllers/sections/info/LocalReferenceNumberController.scala [hunk 5]
```diff
@@ -88,6 +89,22 @@ class LocalReferenceNumberController @Inject()(
         Future.successful(Redirect(controllers.sections.info.routes.DeferredMovementController.onPageLoad(request.ern)))
     }
 
+  private def withDispatchPlaceAnswer(f: DispatchPlace => Future[Result])(implicit request: UserRequest[_]): Future[Result] =
```
Comment (james-work-account) [inline app/controllers/sections/info/LocalReferenceNumberController.scala:92]: I don't like this. As far as I know, the Dispatch Place question should only be asked to XIWK people. It is intentional that no other trader types are asked this question.

**Review by LadonJackson — APPROVED** [review]

**Review by LadonJackson — APPROVED** [review]

**Pull request merged**